armi/reactor/assemblies.py

```diff
@@ -93,11 +68,13 @@ def __init__(self, typ, assemNum=None):
             Name of assembly design (e.g. the name from the blueprints input file).
 
         assemNum : int, optional
-            The unique ID number of this assembly. If none is passed, the class-level
-            value will be taken and then incremented.
+            The unique ID number of this assembly. If None is provided, we generate a
+            random int. This makes it clear that it is a placeholder. When an assembly with
+            a negative ID is placed into a Reactor, it will be given a new, positive ID.
         """
+        # If no assembly number is provided, generate a random number as a placeholder.
         if assemNum is None:
-            assemNum = incrementAssemNum()
+            assemNum = randint(-9e12, -1)
```

Comment:
Is there any reason to make this value a truly random integer? It seems like the fact that this is a placeholder is signified by the sign, not the randomness of the number. Generating a random number is unnecessarily expensive if we could instead just always pass a value of `-1`, for instance.

Reply:
Hmm... this logic is used in a method called "make unique".  So I wanted to err on making each assembly _more_ unique.

You're right, of course. Much of the logic that exists when we add the Assembly to the Reactor would be the same if we used `-1`, or any negative number.

Reply:
Hey @john-science, this random number assignment for `makeUnique` is causing some regression test failures in OXBOW shuffling models. Could you help us understand what drove you towards random int assignment? Why not assign each assembly a unique and valid ID on instantiation (or when `makeUnique` is called) instead of overwriting the random negative int when the assembly is shuffled into the core?

cc @jeffbaylor @opotowsky